packages/pigeon/example/README.md

```diff
@@ -1,155 +1,277 @@
+<?code-excerpt path-base="excerpts/packages/pigeon_example"?>
 # Pigeon Examples
 
+The examples here will cover basic usage. For a more thorough set of examples,
+check the [core_tests pigeon file](../pigeons/core_tests.dart) and 
+[platform test folder](../platform_tests/) ([shared_test_plugin_code](../platform_tests/shared_test_plugin_code/) and [alternate_language_test_plugin](../platform_tests/alternate_language_test_plugin/) especially).
+
+## Invocation
+
+Begin by configuring pigeon at the top of the `.dart` input file.
+In actual use, you would include only the languages
+needed for your project.
+
+<?code-excerpt "../../app/pigeons/messages.dart (config)"?>
+```dart
+@ConfigurePigeon(PigeonOptions(
+  dartOut: 'lib/src/messages.g.dart',
+  dartOptions: DartOptions(),
+  cppOptions: CppOptions(namespace: 'pigeon_example'),
+  cppHeaderOut: 'windows/runner/messages.g.h',
+  cppSourceOut: 'windows/runner/messages.g.cpp',
+  kotlinOut:
+      'android/app/src/main/kotlin/dev/flutter/pigeon_example_app/Messages.g.kt',
+  kotlinOptions: KotlinOptions(),
+  javaOut: 'android/app/src/main/java/io/flutter/plugins/Messages.java',
+  javaOptions: JavaOptions(),
+  swiftOut: 'ios/Runner/Messages.g.swift',
+  swiftOptions: SwiftOptions(),
+  objcHeaderOut: 'macos/Runner/messages.h',
+  objcSourceOut: 'macos/Runner/messages.m',
+  objcOptions: ObjcOptions(prefix: 'PGN'),
```

Comment:
To somewhat reduce the chance of class collisions due to people just blindly copy/pasting, lets add a comment above this line:

```
// Set this to a unique prefix for your plugin or application, per Objective-C naming conventions.
```